pynhm/preprocess/cbh_utils.py

```diff
@@ -0,0 +1,335 @@
+        '_FillValue':9.96921e36,
+        'long_name':'Daily mean relative humidity',
+        'units':'percent',
+        'standard_name':'rhavg',}}
+
+cbh_units = {key:val['units'] for key, val in cbh_metadata.items() if 'units' in list(val.keys())}
+
+created_line = 'Created '
+written_line = 'Written '
+hash_line = '##############'
+hash_line_official = '########################################'
+
+
+def _get_std_name(name: str) -> str:
+    for key, val in cbh_std_name_from_dict.items():
+        if name in val:
+            return key
+    msg = f"The variable name '{name}' can not be mapped to a standard name."
+    raise ValueError(msg)
+
+
+def _cbh_file_to_df(the_file: file_type) -> pd.DataFrame:
+    # This is attempting to handle as many input formats for these kinds of files
+    # as we can find. It may not be comprehensive. See tests for what is currently
+    # handled
+    # Handle netcdf?
+    meta_lines = []
+    with open(the_file, 'r') as file_open:
+        wh_hash_line = -1
+        the_line = ''
+        while hash_line not in the_line:
+            wh_hash_line += 1
+            the_line = file_open.readline()
+            if (
+                    ('//' not in the_line)
+                    and (created_line not in the_line)
+                    and (written_line not in the_line)
+                    and (hash_line not in the_line)):
+                meta_lines += [the_line.strip()]
+
+    col_names = []
+    var_count_dict = {}
+    for posn in range(len(meta_lines)):
+        key, count = meta_lines[posn].split(' ')
+        count = int(count)
+        zs = math.ceil(math.log(count, 10))
+        std_name = _get_std_name(key)
+        col_names += [f"{std_name}{str(ii).zfill(zs)}" for ii in range(count)]
+        # col_names += [f"{key}{str(ii).zfill(zs)}" for ii in range(count)]
+        var_count_dict[key] = count
+
+    # Can we get the hru info? is that standarized at all?
+
+    dtypes = (['str'] * 6) + (['float64'] * len(col_names))
+    date_cols = ["Y", "m", "d", "H", "M", "S"]
+    col_names = date_cols + col_names
+    assert len(dtypes) == len(col_names)
+    dtype_dict = dict(zip(col_names, dtypes))
+    assert len(dtype_dict) == len(col_names)
+
+    # Some files erroneously specify the number of columns
+    # catch that here
+    data = pd.read_csv(
+        the_file,
+        nrows=1,
+        # names=col_names,  # dont specify names
+        header=None,  # dont use default header from first line
+        index_col=False,
+        skiprows=wh_hash_line+1,
+        delim_whitespace=True, dtype=dtype_dict)
+    msg = f"Number of actual data columns does not match metadata info: {meta_lines}"
+    assert len(data.columns) == len(col_names), msg
+    # JLM: is the above sufficient?
+
+    data = pd.read_csv(
+        the_file,
+        names=col_names,
+        index_col=False,
+        skiprows=wh_hash_line+1,
+        delim_whitespace=True, dtype=dtype_dict)
+
+    data['date'] = pd.to_datetime(
```

Comment:
@jdhughes-usgs In youre previous comment, did you mean concatenating the date strings? I think it's readable and vectorized. not too bad. I think you mentioned that passing the magic stuff to pd.read_csv was not future proof.
There's also doing pd.read_csv twice, but that is to catch garbage ascii files (my humble opinion).